bin/functions.sh

```diff
@@ -0,0 +1,56 @@
+# minimum required node version
+REQUIRED_NODE_MAJOR=10
+REQUIRED_NODE_MINOR=13
+
+# minimum required npm version
+REQUIRED_NPM_MAJOR=5
+REQUIRED_NPM_MINOR=5
+
+pecho() { printf %s\\n "$*"; }
+log() { pecho "$@"; }
+error() { log "ERROR: $@" >&2; }
+fatal() { error "$@"; exit 1; }
+is_cmd() { command -v "$@" >/dev/null 2>&1; }
+
+compute_node_args() {
+  NODE_VERSION_STRING=$(node --version)
+  NODE_VERSION_STRING=${NODE_VERSION_STRING#"v"}
+  DETECTED_MAJOR=$(pecho "$NODE_VERSION_STRING" | cut -s -d "." -f 1)
+  ARGS=""
+
+  [ -n "$DETECTED_MAJOR" ] || fatal "Cannot extract $PROGRAM_LABEL major version from version string \"$VERSION_STRING\""
```

Comment:
This logic is repeated in `require_minimal_version`. Please move it to a separate function that is called by both `compute_node_args` and `require_minimal_version`.

Reply:
pushed 5aaeddf to improve this logic